cypress/integration/tests/applicationinventory/application/pagination.test.ts

```diff
@@ -0,0 +1,199 @@
+        Cypress.Cookies.preserveOnce("AUTH_SESSION_ID", "KEYCLOAK_SESSION");
+
+        // Interceptors for Applications
+        cy.intercept("GET", "/api/application-inventory/application*").as("getApplications");
+    });
+
+    after("Perform test data clean up", function () {
+        // Delete the Applications created before the tests
+        if (applicationsList.length > 0) {
+            // Navigate to Application inventory tab
+            clickByText(navMenu, applicationinventory);
+            cy.wait("@getApplications");
+            selectItemsPerPage(100);
+            cy.wait(2000);
+
+            applicationsList.forEach(function (application) {
+                cy.get("td[data-label=Name]").each(($rows) => {
+                    if ($rows.text() === application.name) application.delete();
+                });
+            });
+        }
+    });
+
+    it("Navigation button validations", function () {
+        // Navigate to Application inventory tab
+        clickByText(navMenu, applicationinventory);
+        cy.wait("@getApplications");
+
+        // select 10 items per page
+        selectItemsPerPage(10);
+        cy.wait("@getApplications");
+
+        // Verify next buttons are enabled as there are more than 11 rows present
+        cy.get(commonView.nextPageButton).each(($nextBtn) => {
+            cy.wrap($nextBtn).should("not.be.disabled");
+        });
+
+        // Verify that previous buttons are disabled being on the first page
+        cy.get(commonView.prevPageButton).each(($previousBtn) => {
+            cy.wrap($previousBtn).should("be.disabled");
+        });
+
+        // Verify that navigation button to last page is enabled
+        cy.get(commonView.lastPageButton).should("not.be.disabled");
+
+        // Verify that navigation button to first page is disabled being on the first page
+        cy.get(commonView.firstPageButton).should("be.disabled");
+
+        // Navigate to next page
+        cy.get(commonView.nextPageButton).eq(0).click();
+        cy.wait("@getApplications");
+
+        // Verify that previous buttons are enabled after moving to next page
+        cy.get(commonView.prevPageButton).each(($previousBtn) => {
+            cy.wrap($previousBtn).should("not.be.disabled");
+        });
+
+        // Verify that navigation button to first page is enabled after moving to next page
+        cy.get(commonView.firstPageButton).should("not.be.disabled");
+    });
+
+    it("Items per page validations", function () {
+        // Navigate to Application inventory tab
+        clickByText(navMenu, applicationinventory);
+        cy.wait("@getApplications");
+
+        // Select 10 items per page
+        selectItemsPerPage(10);
+        cy.wait(2000);
+
+        // Verify that only 10 items are displayed
+        cy.get("td[data-label=Name]").then(($rows) => {
+            cy.wrap($rows.length).should("eq", 10);
+        });
```

Comment:
@ganeshhubale I suggest changing the code so we select only the main rows and not the expanded rows of the table.

```suggestion
       cy.get(".pf-c-table > tbody > tr")
            .not(".pf-c-table__expandable-row")
            .find("td[data-label=Name]")
            .then(($rows) => {
                cy.wrap($rows.length).should("eq", 10);
            });
```